src/app/marketplace/edit-ad.component.ts

```diff
@@ -0,0 +1,329 @@
+   * a new ad based on the current one
+   */
+  save(insertNew?: boolean) {
+
+    validateBeforeSubmit(this.form);
+    if (!this.form.valid) {
+      return;
+    }
+
+    const value = this.form.value;
+
+    value.publicationPeriod = this.ApiHelper.dateRange(value.publicationBeginDate, value.publicationEndDate);
+    delete value['publicationBeginDate'];
+    delete value['publicationEndDate'];
+
+    value.promotionalPeriod = this.ApiHelper.dateRange(value.promotionalBeginDate, value.promotionalEndDate);
+    delete value['promotionalBeginDate'];
+    delete value['promotionalEndDate'];
+
+    const onFinish: any = (id: string) => {
+      this.notification.snackBar(this.i18n.ad.adSaved);
+      if (insertNew) {
+        EditAdComponent.BASED_ON_ID = id || this.id;
+        if (this.create) {
+          this.reload();
+        } else {
+          this.router.navigate(['/marketplace', this.owner, this.data.kind, 'new'], { replaceUrl: true });
+        }
+      } else if (this.basedOnId) {
+        this.router.navigate(['/marketplace', 'edit', id], { replaceUrl: true });
+      } else {
+        history.back();
+      }
+    };
+
+    if (this.create) {
+
+      value.images = this.uploadedImages.map(i => i.id);
+
+      if (this.data.requiresAuthorization) {
+        // When requires authorization submit as draft when saving for first time
+        value.submitForAuthorization = false;
+      }
+
+      this.addSub(this.marketplaceService.createAd({
+        user: this.owner,
+        body: value
+      }).subscribe(onFinish));
+
+    } else {
+
+      const updateAdReq = this.marketplaceService.updateAd({
+        ad: this.id,
+        body: value
+      });
+
+      // If the main image has changed reload the ad version
+      if (this.mainImageChanged()) {
+        this.addSub(this.marketplaceService.getAdDataForEdit({ ad: this.id, fields: ['advertisement.version'] })
+          .subscribe(data => {
+            value.version = data.advertisement.version;
+            this.addSub(updateAdReq.subscribe(onFinish));
+          }));
+      } else {
+        this.addSub(updateAdReq.subscribe(onFinish));
+      }
+    }
+  }
+
+  /**
+   * Returns if the main ad image has changed
+   */
+  protected mainImageChanged(): boolean {
+    const currentImage = this.images.length > 0 ? this.ApiHelper.internalNameOrId(this.images[0]) : null;
+    // Check any change in the main image, either added and removed, or changed
+    return this.mainImage == null && currentImage == null && this.hasRemovedImages ||
+      this.mainImage !== currentImage;
+  }
+
+  /**
+   * Reorder or remove images
+   */
+  manageImages() {
+    const ref = this.modal.show(ManageImagesComponent, {
+      class: 'modal-form',
+      initialState: {
+        images: this.images,
+        manageAfterConfirm: this.create
+      }
+    });
+    const component = ref.content as ManageImagesComponent;
+    this.addSub(component.result.pipe(take(1)).subscribe(result => {
```

Comment:
You don't need the `addSub` (which is used to immediately unsubscribe) if you use `take(1)` (or, would be better, `first`), because it automatically unsubscribes after the first result is emitted.